authors/apps/authentication/tests/test_user_registration.py

```diff
@@ -0,0 +1,139 @@
+        self.signup_url = reverse('authentication:auth-register')
+        self.signup_data = {
+            "user": {
+                "username": "gigz",
+                "email": "jake@jake.jake",
+                "password": "jakejake"
+                }}
+        self.login_data = {
+            "user": {
+                "email": "jake@jake.jake",
+                "password": "jakejake"
+                }}
+        self.login_unregistered_user_data = {
+            "user": {
+                "email": "muinde@jake.jake",
+                "password": "jakejake"
+                }}
+        self.signup_empty_username = {
+            "user": {
+                "username": "",
+                "email": "jake@jake.jake",
+                "password": "jakejake"
+                }}
+        self.signup_empty_email = {
+            "user": {
+                "username": "gigz",
+                "email": "",
+                "password": "jakejake"
+                }}
+        self.signup_empty_password = {
+            "user": {
+                "username": "gigz",
+                "email": "jake@andela.com",
+                "password": ""
+                }}
+        self.signup1 = {
+            "user": {
+                "username": "gigz",
+                "email": "jake@andela.com",
+                "password": "jakejake"
+                }}
+        self.signup2 = {
+            "user": {
+                "username": "jake",
+                "email": "jake@andela.com",
+                "password": "jakejake"
+                }}
+        self.signup3 = {
+            "user": {
+                "username": "jake",
+                "email": "jakey@andela.com",
+                "password": "jakejake"
+                }}
+        self.duplicate_email = {
+            "user": {
+                "username": "jake",
+                "email": "jacky@andela.com",
+                "password": "jakejake"
+                }}
+        self.duplicate_email2 = {
+            "user": {
+                "username": "jacky2",
+                "email": "jacky@andela.com",
+                "password": "jakejake"
+                }}
+
+    def test_register_user(self):
+        ''''Test register user'''
+        response = self.client.post(self.signup_url, self.signup_data,
+                                    format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.content, {"user": {"email": "jake@jake.jake",
+                                            "username": "gigz"}})
+
+    def test_empty_username(self):
+        '''Test register user with empty username'''
+        response = self.client.post(self.signup_url,
+                                    self.signup_empty_username,
+                                    format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(json.loads(response.content), {"errors": {"username":
```

Comment:
why do we have to use json.loads here and not on line 82?

Reply:
They are two different tests the one on line 82 returns serialized data and the one on line 91  returns an error message.